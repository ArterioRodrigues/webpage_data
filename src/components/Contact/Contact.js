import React from 'react';
import contactInfo from '../../data/contact.json';

const Contact = () => (
  <div className="container">
    <div className="contact">
      <h2 className="container__title">contact</h2>
      <h3 className="container__subtitle">Feel free to contact me</h3>
      <div className="container__list">
        <ul>
          {contactInfo.map(contactItem => (
            <li key={`contact-page-${contactItem.name}`}>
              <a
                href={contactItem.url}
                aria-label={`Find me on ${contactItem.name}`}
                title={`Find me on ${contactItem.name}`}
                target="_blank"
                rel="noopener noreferrer"
              >
                {contactItem.name}
              </a>
            </li>
          ))}
        </ul>
      </div>
    </div>
  </div>
);

export default Contact;
