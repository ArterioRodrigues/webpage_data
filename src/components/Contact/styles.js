/**
 * Contact component styles.
 */
import { mSize } from '../../theme';

export default () => ({
  contactItem: {
    fontSize: mSize,
    marginBottom: '10px',
  },
});
