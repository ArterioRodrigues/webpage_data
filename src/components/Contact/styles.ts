/**
 * Contact component styles.
 */
import { Styles } from 'react-jss';
import { mSize } from '../../theme';

export default {
  contactItem: {
    fontSize: mSize,
    marginBottom: '10px',
  },
} as Styles;
