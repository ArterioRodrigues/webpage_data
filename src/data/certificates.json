{
  "sectionTitle": "certifications",
  "sectionSubtitle": "Certificates that I have collected over time",
  "content": [
    {
      "name": "Responsive Web Design Certification",
      "url": "https://www.freecodecamp.org/certification/caglarturali/responsive-web-design"
    },
    {
      "name": "Introduction to Game Development",
      "url": "https://www.coursera.org/account/accomplishments/certificate/GKDBNAGKPD6M"
    },
    {
      "name": "Introduction to Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/UC3L666SD5TG"
    },
    {
      "name": "3D Models for Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/4YHK3LYWMJPV"
    },
    {
      "name": "The Art of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/DNVYLQQL3WVF"
    },
    {
      "name": "The Technology of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/VZDUNEC24XER"
    },
    {
      "name": "Pro Tools Basics",
      "url": "https://www.coursera.org/account/accomplishments/certificate/YW4QY26UHC5F"
    }
  ]
}
