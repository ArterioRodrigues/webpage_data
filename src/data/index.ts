export * from './certificates';
export * from './contact';
export * from './education';
export * from './greeter';
export * from './metadata';
export * from './projects';
export * from './skills';
