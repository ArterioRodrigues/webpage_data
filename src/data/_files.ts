import { NavLink } from '../models';
import Greeter from '../views/Greeter';
import Skills from '../views/Skills';
import Education from '../views/Education';
import Projects from '../views/Projects';
import Certificates from '../views/Certificates';
import Contact from '../views/Contact';
import { certificates, contact, education, projects, skills } from '.';

export const files: NavLink[] = [
  {
    icon: ['fab', 'html5'],
    color: '#e44d26',
    url: '/',
    name: 'index.html',
    route: {
      component: Greeter,
      data: contact,
    },
  },
  {
    icon: ['fab', 'js'],
    color: '#ffca28',
    url: '/skills',
    name: 'skills.js',
    route: {
      component: Skills,
      data: skills,
    },
  },
  {
    icon: 'university',
    color: '#42a5f5',
    url: '/education',
    name: '.educationrc',
    route: {
      component: Education,
      data: education,
    },
  },
  {
    icon: 'sliders-h',
    color: '#98c379',
    url: '/projects',
    name: 'projects.config',
    route: {
      component: Projects,
      data: projects,
    },
  },
  {
    icon: ['fab', 'sass'],
    color: '#e13e76',
    url: '/certifications',
    name: 'certifications.sass',
    route: {
      component: Certificates,
      data: certificates,
    },
  },
  {
    icon: ['fab', 'markdown'],
    color: '#abb2bf',
    url: '/contact',
    name: 'contact.md',
    route: {
      component: Contact,
      data: contact,
    },
  },
];
