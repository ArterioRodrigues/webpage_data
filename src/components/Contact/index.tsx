/**
 * Contact component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import Container from '../Container';
import { ContactItem, Page } from '../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface ContactProps {
  data: Page<ContactItem>;
}

const Contact: React.FC<ContactProps> = ({ data }) => {
  const classes = useStyles();

  return (
    <Container title={data.name} subtitle={data.desc}>
      <ul>
        {data.content.map(({ name, url }) => (
          <li className={classes.contactItem} key={`contact-page-${name}`}>
            <a
              href={url}
              aria-label={`Find me on ${name}`}
              title={`Find me on ${name}`}
              target="_blank"
              rel="noopener noreferrer"
            >
              {name}
            </a>
          </li>
        ))}
      </ul>
    </Container>
  );
};

export default Contact;
