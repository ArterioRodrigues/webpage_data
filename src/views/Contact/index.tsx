/**
 * Contact component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import Container from '../../components/Container';
import { ContactItem, BaseProps } from '../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export type ContactProps = BaseProps<ContactItem>;

const Contact: React.FC<ContactProps> = ({ data, fileName }) => {
  const classes = useStyles();

  return (
    <Container
      title={data.name}
      subtitle={data.desc}
      seo={{ title: fileName, description: data.desc }}
    >
      <ul>
        {data.content.map(({ name, url }) => (
          <li className={classes.contactItem} key={`contact-page-${name}`}>
            <a
              href={url}
              aria-label={`Find me on ${name}`}
              title={`Find me on ${name}`}
              target="_blank"
              rel="noopener noreferrer"
            >
              {name}
            </a>
          </li>
        ))}
      </ul>
    </Container>
  );
};

export default Contact;
