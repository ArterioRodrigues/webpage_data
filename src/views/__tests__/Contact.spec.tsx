import React from 'react';
import { render } from '@testing-library/react';
import Contact from '../Contact';
import { contact } from '../../data';

describe('<Contact />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <Contact data={contact} fileName="contact.test" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
