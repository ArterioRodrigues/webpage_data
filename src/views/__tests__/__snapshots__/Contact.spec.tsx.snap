// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root-0-2-2"
  >
    <h2
      class="title-0-2-3"
    >
      contact
    </h2>
    <h3
      class="subtitle-0-2-4"
    >
      Feel free to contact me
    </h3>
    <div
      class="content-0-2-5"
    >
      <ul>
        <li
          class="contactItem-0-2-1"
        >
          <a
            aria-label="Find me on Telegram"
            href="https://t.me/caglarturali"
            rel="noopener noreferrer"
            target="_blank"
            title="Find me on Telegram"
          >
            Telegram
          </a>
        </li>
        <li
          class="contactItem-0-2-1"
        >
          <a
            aria-label="Find me on GitHub"
            href="https://github.com/caglarturali"
            rel="noopener noreferrer"
            target="_blank"
            title="Find me on GitHub"
          >
            GitHub
          </a>
        </li>
        <li
          class="contactItem-0-2-1"
        >
          <a
            aria-label="Find me on LinkedIn"
            href="https://linkedin.com/in/caglarturali"
            rel="noopener noreferrer"
            target="_blank"
            title="Find me on LinkedIn"
          >
            LinkedIn
          </a>
        </li>
        <li
          class="contactItem-0-2-1"
        >
          <a
            aria-label="Find me on Instagram"
            href="https://www.instagram.com/caglarturali/"
            rel="noopener noreferrer"
            target="_blank"
            title="Find me on Instagram"
          >
            Instagram
          </a>
        </li>
        <li
          class="contactItem-0-2-1"
        >
          <a
            aria-label="Find me on goodreads"
            href="https://www.goodreads.com/caglarturali"
            rel="noopener noreferrer"
            target="_blank"
            title="Find me on goodreads"
          >
            goodreads
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
